content/concepts/the-rippled-server/rippled-server-modes.md

```diff
@@ -2,61 +2,83 @@
-* It could selectively show or hide payment paths and currency exchange offers to guarantee its own profit while not providing you the best deal.
-* If you sent it your address's secret key, it could make arbitrary transactions on your behalf, and even transfer or destroy all the money your address holds.
+- [Validator](#validators)
+- [API Server](#api-servers)
+- [Public Hubs](#public-hubs)
 
-Additionally, running your own server gives you admin control over it, which allows you to run important admin-only and load-intensive commands. If you use a shared server, you have to worry about other users of the same server competing with you for the server's computing power. Many of the commands in the WebSocket API can put a lot of strain on the server, so `rippled` has the option to scale back its responses when it needs to. If you share a server with others, you may not always get the best results possible.
+Network mode servers connect to [Mainnet](parallel-networks.html) by default.
+
+
+### API Servers
+
+All network mode servers provide [APIs](rippled-api.html) for purposes like submitting transactions, checking balances and settings, and administering the server. If you query the XRP Ledger for data or submit transactions for business use, it can be useful to [run your own server](the-rippled-server.html#reasons-to-run-your-own-server).
+
+#### Full History Servers
+
+Unlike some other blockchains, the XRP Ledger does not require servers to have a complete transaction history to know the current state and process new transactions. As a server operator, you decide how much [ledger history](ledger-history.html) to store at a time. However, a network mode server can only answer API requests using the ledger history it has locally available. For example, if you keep six months of history, your server can't describe the outcome of a transaction from a year ago. API servers with [full history](ledger-history.html#full-history) can report all transactions and balances since the inception of the XRP Ledger.
 
-Finally, if you run a validating server, you can use a stock server as a proxy to the public network while keeping your validating server on a private network only accessible to the outside world through the stock server. This makes it more difficult to compromise the integrity of your validating server.
 
 ### Public Hubs
 
-A public hub is a stock server with lots of [peer protocol connections](peer-protocol.html) to other servers. You can help the XRP Ledger network maintain efficient connectivity by running a stock server as a public hub. Successful public hubs embody the following traits:
+A hub server is a network mode server with lots of [peer protocol connections](peer-protocol.html) to other servers. Hub servers, especially _public hubs_ that allow connections from the open internet, help the XRP Ledger network maintain efficient connectivity. Successful public hubs embody the following traits:
 
 - Good bandwidth.
```

Comment:
trivial nit: I don't think the periods are required here since these are just phrases, not complete sentences.

Reply:
That's true, but I vaguely remember us making a rule of thumb that we should always end list items and stuff with periods for consistency.